Hunting Queries/MultipleDataSources/MultiplePasswordResetbyUser.txt

```diff
@@ -0,0 +1,61 @@
+// Name: Multiple Password Reset by user
+// Description: This query will determine multiple password resets by user across multiple data sources. 
+// Account manipulation including password reset may aid adversaries in maintaining access to credentials 
+// and certain permission levels within an environment.
+// 
+// Id: 0b9ae89d-8cad-461c-808f-0494f70ad5c4
+//
+// DataSource: #AuditLogs, #Syslog, #SecurityEvent, #OfficeActivity
+//
+// Tactics: #Discovery, #CredentialAccess
+//
+let timeframe = 1d;
+let PerUserThreshold = 5;
+let TotalThreshold = 100;
+let PasswordResetMultiDataSource =
+(union isfuzzy=truegit
+(
+//WEF Password reset events
+//4723: An attempt was made to change an account's password
+//4724: An attempt was made to reset an accounts password
+SecurityEvent
+| where TimeGenerated >= ago(timeframe)
+| where EventID in ("4723","4724")
+| extend DataSource="WEF"
+| project DataSource, TimeGenerated,AccountType,Account
+),
+(
+//Azure Active Directory Password reset events
+AuditLogs
+| where Category == "UserManagement" and Result == "success"
+| where TimeGenerated >= ago(timeframe)
+| where OperationName in ("Change user password","Reset user password","Change password (self-service)","Reset password (self-service)")
+| extend DataSource="Azure Active Directory"
+| extend AccountType = tostring(TargetResources[0].type, Account = tostring(TargetResources[0].userPrincipalName))
```

Comment:
If I run this section by itself, it fails.I think you are missing a closing paren before Account =, then remove the closing paren after ".userPrincipalName))".  Also, there is no computer Field.  this works, just change the ago back to "timeframe" -

AuditLogs
| where Category == "UserManagement" and Result == "success"
| where TimeGenerated >= ago(1d)
| where OperationName in ("Change user password","Reset user password","Change password (self-service)","Reset password (self-service)")
| extend DataSource="Azure Active Directory"
| extend AccountType = tostring(TargetResources[0].type), Account = tostring(TargetResources[0].userPrincipalName)
| project DataSource, TimeGenerated,AccountType,Account

Reply:
submitted new commit with changes